Data/Text.hs

```diff
@@ -1,4 +1,4 @@
-{-# LANGUAGE BangPatterns, CPP, MagicHash, Rank2Types, UnboxedTuples #-}
```

Comment:
Requiring full TH to build `text` is not acceptable; however, `TemplateHaskellQuotes` would be acceptable

Reply:
This is awkward. `TemplateHaskellQuotes` is only since 8.0, and I'm drawing a blank on how to write this without quote syntax: `mkName` and `lookupValueName` operate at the use site, which is unsuitable (what if users don't import `Data.Text`? What if it's been renamed?).

Might have to put this on ice until 7.10 and lower are out of the support window.

Reply:
Well, if you can make it work for GHC >= 8 with THQ, we can conditionally enable old-style TH for GHC < 8 . We just need to make sure that recentish GHCs don't require to have a bytecode interpreter in order to build the core library `text`